### PR TITLE
Show remote branches and tags for completion of git-refs

### PR DESCRIPTION
**Pull request by hexium310**
This PR fixes the completion using git-refs like `git fetch origin **` to show the branches and the tags on remote. Bedsides the bare git repository is added for testing.

### CODE REVIEW

File: tests/git.zunit [hunk 1]
```diff
@@ -31,7 +31,7 @@
         git clean -d -f
     } &> /dev/null
 
-    rm -rf .git
```
Comment (chitoku-k) [inline tests/git.zunit:34]: Good catch!

File: tests/git.zunit [hunk 2]
```diff
@@ -3535,10 +3535,48 @@
     _fzf_complete_git 'git fetch '
 }
 
+# FIXME: the error that occurs in mocks isn't shown.
```
Comment (hexium310) [inline tests/git.zunit:3538]: The cause is that AWK deletes the error.

File: src/completers/git.zsh [hunk 3]
```diff
@@ -834,11 +834,20 @@ _fzf_complete_git-refs() {
...
     shift
 
-    local ref=${${$(git config remote.$repository.fetch 2> /dev/null)#*:}%\*}
-
     _fzf_complete --ansi --tiebreak=index ${(Q)${(Z+n+)fzf_options}} -- $@$prefix_option$prefix_ref < <(
-        git for-each-ref "$ref" --format='%(refname:short) %(contents:subject)' 2> /dev/null |
-            _fzf_complete_tabularize ${fg[yellow]}
+        git ls-remote --quiet --heads --tags "$repository" 2> /dev/null | awk '
+            {
+                if(match($2, /refs\/heads\//)) {
```
Comment (chitoku-k) [inline src/completers/git.zsh:840]:
```suggestion
                if ($2 ~ /refs\/heads\//) {
```

File: src/completers/git.zsh [hunk 4]
```diff
@@ -834,11 +834,20 @@ _fzf_complete_git-refs() {
...
+                    kind = "branch"
+                } else if(match($2, /refs\/tags\//)) {
```
Comment (chitoku-k) [inline src/completers/git.zsh:842]:
```suggestion
                } else if ($2 ~ /refs\/tags\//) {
```

Comment (chitoku-k) [inline src/completers/git.zsh:842]:
```suggestion
                if (match($2, /refs\/heads\//)) {
                    kind = "branch"
                } else if (match($2, /refs\/tags\//)) {
```

File: tests/_support/remote.git/info/exclude [hunk 5]
```diff
@@ -0,0 +1,6 @@
+# git ls-files --others --exclude-from=.git/info/exclude
+# Lines that start with '#' are comments.
+# For a project mostly in C, the following would be a good set of
+# exclude patterns (uncomment them if you want to use them):
+# *.[oa]
+# *~
```
Comment (chitoku-k) [inline tests/_support/remote.git/info/exclude:6]: This file does not have to be committed.

File: src/completers/git.zsh [hunk 6]
```diff
@@ -834,11 +834,20 @@ _fzf_complete_git-refs() {
...
+                    kind = "tag"
+                } else {
+                    next
+                }
+
+                print substr($2, RSTART + RLENGTH), kind
+            }
```
Comment (chitoku-k) [inline src/completers/git.zsh:849]:
```suggestion
            match($2, /refs\/heads\//) {
                print substr($2, RSTART + RLENGTH), "branch"
            }
            match($2, /refs\/tags\//)) {
                print substr($2, RSTART + RLENGTH), "tag"
            }
```

**Review by chitoku-k — APPROVED** [review]
Perfect!

**Pull request merged**